src/core/util/mod.rs

```diff
@@ -0,0 +1,78 @@
+#[cfg(feature = "reqwest")]
+use anyhow::Context;
+use anyhow::Result;
+use async_trait::async_trait;
+use http::{Request, Response};
+
+/// Generic HTTP client.
+///
+/// A trait is used here so to facilitate native HTTP/TLS when compiled for mobile applications.
+#[async_trait]
+pub trait HttpClient {
```

Comment:
I think it would be good to rename this `AsyncHttpClient` to avoid any confusion